[IMP] Evaluate web_timeline colors attribute as python

karlis-dreizis opened the pull request:
I'd like to get some feedback on this change.

Con: its a backwards incompatible change, currently colors predicates are evaluated as javascript. I would have to update project_timeline and any other modules with dependencies on web_timeline. 

Pro: this enables the use of record attributes on the rhs of the expression, logical operators, etc.

Use case: project_timeline, project.task red: date_deadline > date_end

karlis-dreizis committed:
[IMP] Evaluate web_timeline colors attribute as python

The js eval implementation treats the right hand side of the expression
as a string. Using py.js to parse the predicate expression allows the
predicate to reference a record value itself.

karlis-dreizis committed:
[IMP] Evaluate web_timeline colors attribute as python

The js eval implementation treats the right hand side of the expression
as a string. Using py.js to parse the predicate expression allows the
predicate to reference a record value itself.

tarteo reviewed (changes requested):
Can you update the readme and increase the version number.

karlis-dreizis committed:
Merge branch '11.0' of https://github.com/InfernalLV/web into 11.0

karlis-dreizis commented:
README.rst updated

Tardo commented on web_timeline/static/src/js/timeline_view.js:
```diff
@@ -84,8 +84,10 @@ odoo.define('web_timeline.TimelineView', function (require) {
             this.parse_colors();
-            for (var i=0; i<this.colors.length; i++) {
-                fieldNames.push(this.colors[i].field);
+            for (let color of this.colors) {
```
Odoo v11.0 doesn't support ES6. Can't use 'let' and 'of' :(
https://github.com/lukehoban/es6features#let--const

karlis-dreizis replied:
Waiting for IE11 to be dropped :)

karlis-dreizis committed:
[IMP] Evaluate web_timeline colors attribute as python

The js eval implementation treats the right hand side of the expression
as a string. Using py.js to parse the predicate expression allows the
predicate to reference a record value itself.

karlis-dreizis commented:
OCA/project#480 brings project_timeline module in line.

karlis-dreizis committed:
Fix version number

Comply with OCA. Address the breaking change in the version number.

karlis-dreizis reviewed (commented):
The `__manifest__.py` file is fixed up

tarteo reviewed (changes requested):
Please check travis for missing semicolons and indentation errors.

karlis-dreizis committed:
Add missing semicolons, use strict equality

Address [Error/semi], [Error/eqeqeq] in changed lines.

karlis-dreizis committed:
Correct indentation

Correct [Error/indent]

tarteo reviewed (approved):
Thanks! Code LGTM
